### PR TITLE
update ui test cases

### PR DESCRIPTION
**Pull request by kurokn**

### CODE REVIEW

File: app/jest/layout.test.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+const ROCKET_CHAT_HYPERLINK = 'https://chat.developer.gov.bc.ca/channel/sso';
+const PATHFINDER_SSO_HYPERLINK = 'mailto:bcgov.sso@gov.bc.ca';
+const DOCUMENTATION_HYPERLINK = 'https://github.com/bcgov/sso-keycloak/wiki';
+const DISCLAIMER_HYPERLINK = 'https://www2.gov.bc.ca/gov/content/home/disclaimer';
+const PRIVACY_HYPERLINK = 'https://www2.gov.bc.ca/gov/content/home/privacy';
+const ACCESSIBILITY_HYPERLINK = 'https://www2.gov.bc.ca/gov/content/home/accessible-government';
+const COPYRIGHT_HYPERLINK = 'https://www2.gov.bc.ca/gov/content/home/copyright';
+
+const session = {
```
Comment (NithinKuruba) [inline app/jest/layout.test.tsx:22]: @kurokn please ignore redundant code, you can make use of session object from `home.test.tsx`

File: app/jest/layout.test.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    expect(screen.getByRole('link', { name: 'Pathfinder SSO' })).toHaveAttribute('href', PATHFINDER_SSO_HYPERLINK);
+    expect(screen.getByRole('link', { name: 'Documentation' })).toHaveAttribute('href', DOCUMENTATION_HYPERLINK);
+    expect(screen.getByRole('link', { name: 'Disclaimer' })).toHaveAttribute('href', DISCLAIMER_HYPERLINK);
+    expect(screen.getByRole('link', { name: 'Privacy' })).toHaveAttribute('href', PRIVACY_HYPERLINK);
+    expect(screen.getByRole('link', { name: 'Accessibility' })).toHaveAttribute('href', ACCESSIBILITY_HYPERLINK);
+    expect(screen.getByRole('link', { name: 'Copyright' })).toHaveAttribute('href', COPYRIGHT_HYPERLINK);
+  });
+
+  it('testing on the My Profile module', async () => {
+    const { asFragment } = render(<LayoutComponent />);
```
Comment (NithinKuruba) [inline app/jest/layout.test.tsx:84]: @kurokn could you brief on this test?? was it intended for `profile` modal

File: app/jest/my-dashboard/my-projects/assignUsersToRoles.test.tsx [hunk 3]
```diff
@@ -50,36 +54,30 @@ jest.mock('services/bceid-webservice', () => ({
...
-    fireEvent.change(searchInput, { target: { value: 'sample_input' } });
-    expect(screen.getByDisplayValue('sample_input')).toBeInTheDocument();
-  });
-
   it('Should be able to list search results, after click on the Search button; Should be able to click the View button for listed search result, and corresponding modal showing up', async () => {
     render(<UserRoles selectedRequest={{ ...sampleRequest, environments: ['dev', 'test'], devIdps: ['idir'] }} />);
-    const searchUserInput = screen.getByRole('textbox');
-    fireEvent.change(await searchUserInput, { target: { value: 'sample_user' } });
+    await waitFor(async () => {
+      fireEvent.click(await screen.findByRole('option', { name: 'Dev' }));
```
Comment (NithinKuruba) [inline app/jest/my-dashboard/my-projects/assignUsersToRoles.test.tsx:60]: @kurokn you need to validate all three environments and IDPs. Each IDP has different options. Example: when you select BCeID, you can choose `Display Name` or `Username`, which you need to validate

File: app/jest/my-dashboard/my-projects/roleManagement.test.tsx [hunk 4]
```diff
@@ -67,120 +73,164 @@ describe('role management tab', () => {
...
+    expect(screen.queryAllByTestId('role-name-input-field')).toHaveLength(2);
+    const removeRoleNameInput = screen.getAllByRole('img', { name: 'Remove Role' });
+    fireEvent.click(removeRoleNameInput[1]);
+    expect(screen.queryAllByTestId('role-name-input-field')).toHaveLength(1);
   });
 
   it('Should be able to input keywords in Create New Role input field', async () => {
-    render(<RoleManagement integration={{ ...sampleRequest }} />);
+    render(<CreateRoleContent integrationId={1} environments={['dev']} />);
+
```
Comment (NithinKuruba) [inline app/jest/my-dashboard/my-projects/roleManagement.test.tsx:107]: @kurokn are you validating `save` button callout when creating a new role??

File: app/jest/layout.test.tsx (hunk 2 above)
Comment (kurokn) [reply app/jest/layout.test.tsx:69]: i tried with different query statement, i.e. findbyXX, waitFor... and using snapshot to see the rendered component,
but the UserProfileModal still not showing up. Need help here, thanks

File: app/jest/my-dashboard/my-projects/assignUsersToRoles.test.tsx (hunk 3 above)
Comment (kurokn) [reply app/jest/my-dashboard/my-projects/assignUsersToRoles.test.tsx]: @NithinKuruba i followed the same path as what you did in the `assignServiceAccountsToRoles.test` and trying to update the dropdown selection value, but seems the fireEvent.click on the `option idp` is not working properly, even the statement passed
current version only renders the component with one specific idp situation only, for checking the associated user-selection-prop

File: app/jest/my-dashboard/my-projects/roleManagement.test.tsx (hunk 4 above)
Comment (NithinKuruba) [reply app/jest/my-dashboard/my-projects/roleManagement.test.tsx:111]: @kurokn the `save` is making a callout to `bulkCreateRole` function at `./sso-requests/app/page-partials/my-dashboard/RoleManagement/CreateRoleContent.tsx` through use of a `ref` and `useImperativeHandle` hooks. So try to mock that function and let me know if that works?

**Review by NithinKuruba — APPROVED** [review]

**Pull request merged**